src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/ExponentialRetry.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.DarcLib
+{
+    public class ExponentialRetry
+    {
+        private static readonly Random Randomizer = new Random();
+
+        private const int RetryCount = 15;
```

Comment:
15 might be high and could lead to timeouts.

Reply:
With the current settings the 15th retry will wait a max of 51 seconds. That could cause timeout issues if we have strict timeouts above it. The total length of all retries is 217 seconds so if we have every call we ever do retry the maximum number of times it might be a problem.

Reply:
This represents 3.6 minutes which by itself won't hit Service Fabric's default timeout of 5 minutes but this doesn't account other operations like cloning the repo, updating files, etc. Also, we made the change to override the timeout to 10 minutes, Still I think 15 retries is too high and we can make it smaller.

Reply:
With a retry count of 10 the maximum total delay is 55 seconds. That might not be enough to avoid some kinds of network failures, but would probably work.

Reply:
Our current retry logic defaults retries to 3 but since now we have a factor based added timeout that as you mention wouldn't go beyond 55 seconds I'm fine with 10.